### PR TITLE
Change container ports and names on E2E test setup

### LINKED ISSUE #1560: E2E: Avoid server containers conflict
[Reference #539](https://github.com/Automattic/woocommerce-payments/pull/539#issuecomment-659338263)

Is a know problem that the e2e tests setup creates a server containers conflict. Because we are cloning the server and using the same docker-compose.yml config. Forcing us to remove the current server containers or modify by hand the e2e server clone.

### 1. Use environment variables in compose (🤓 elegant one)
Update server `docker-compose.yml` to use env vars with fallback values like:
```yml
container_name: ${COMPOSE_WP_NAME:-woocommerce_payments_server_wordpress}
ports:
  - "${COMPOSE_WP_PORT:-8086}:80"
```
And update `tests/e2e/README.md` to include the new variables to define before running setup.

### 2. Patch compose (🔨 hacky one)
Patch the `docker-compose.yml` file in `tests/e2e/env/setup.sh` after clone using something like:
```sh
perl -i -pe 's/(container_name:.*)/$1_e2e/' docker-compose.yml
perl -i -pe 's/8086/8089/' docker-compose.yml
```

### PR DESCRIPTION
**Pull request by tpaksu**
Fixes #1560

#### Changes proposed in this Pull Request

Modify docker-compose.yml for WCPay containers on E2E test setup

E2E test setup command `npm run test:e2e-setup` prepares both server and client instances to test the system as a whole, and to create the server environment, it directly clones the WCPay server repository and runs the docker-compose routine.

While we also develop the server code, our machines have the containers running most of the times, and to run E2E tests, we need to stop the development containers and then run the E2E containers to prevent the port and name conflicts.

To overcome this problem, we need to separate the development environment and the E2E environment, so we will use a different configuration override file to change the default ports and container names while preparing the E2E test setup. 

This PR provides that.

#### Testing instructions

- Clone the WCPay server repo
- On the root folder, do the installation as `README.md` file explains, and run the containers with `npm run up`
- Clone the WCPay client repo
- On the root folder, do the installation as `README.md` file explains
- Run `npm run test:e2e-setup

**BEFORE:**
```
Creating woocommerce_payments_server_mysql      ... error
Creating woocommerce_payments_server_phpmyadmin ... 

ERROR: for woocommerce_payments_server_mysql  Cannot create container for service db: Conflict. The container name "/woocommerce_payments_server_mysql" is already in use by container "ed023817c4b87d7aec59b2c561d674d716f2391c9148b968aCreating woocommerce_payments_server_phpmyadmin ... error

ERROR: for woocommerce_payments_server_phpmyadmin  Cannot create container for service phpMyAdmin: Conflict. The container name "/woocommerce_payments_server_phpmyadmin" is already in use by container "cdc777add30e0e5eb97d0666addd34f7a094063ad0d5ae290d21e6f8179458fd". You have to remove (or rename) that container to be able to reuse that name.

ERROR: for db  Cannot create container for service db: Conflict. The container name "/woocommerce_payments_server_mysql" is already in use by container "ed023817c4b87d7aec59b2c561d674d716f2391c9148b968ae10ec9f28b87a1e". You have to remove (or rename) that container to be able to reuse that name.

ERROR: for phpMyAdmin  Cannot create container for service phpMyAdmin: Conflict. The container name "/woocommerce_payments_server_phpmyadmin" is already in use by container "cdc777add30e0e5eb97d0666addd34f7a094063ad0d5ae290d21e6f8179458fd". You have to remove (or rename) that container to be able to reuse that name.
ERROR: Encountered errors while bringing up the project.
```

**AFTER:**

```
Recreating woocommerce_payments_server_mysql_e2e      ... done
Recreating woocommerce_payments_server_phpmyadmin_e2e ... done
Recreating woocommerce_payments_server_wordpress_e2e  ... done
```

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

**Comment by tpaksu** [comment]
The E2E tests are failing because server code isn't merged yet. After the merge, I should re-trigger the tests with a dummy commit to check everything is in place. Until that, don't merge this please 😁

**Comment by dmallory42** [comment]
> After the merge, I should re-trigger the tests with a dummy commit to check everything is in place

FYI you can do this by clicking on the `Details` button next to the failing test and then clicking `Re-Run Jobs`, so no need for another commit. 🙂

**Review by ismaeldcom — APPROVED** [review]
@tpaksu Thanks for working patiently on this and provide an excellent solution! 🎉 

Everything is working as expected. Let's ship it! 🚢

**Pull request merged**
